fix(ai-gateway): Align providers docs with 0.0.47 API schema (#5875)
Kong/developer.konghq.com: app/_ai_gateway_entities/ai-provider.md

    url: /ai-gateway/entities/ai-model/
  - text: AI Policy entity
    url: /ai-gateway/entities/ai-policy/
  - text: AI Identity Provider entity
    url: /ai-gateway/entities/ai-identity-provider/
faqs:
  - q: What happens when I update an AI Model Provider's credentials?
    a: |
      {{site.ai_gateway}} propagates the credential change to every AI Model that references the
      AI Model Provider (by `name` or `id`). The next request through any of those AI Models uses the updated
      credentials.

  - q: How does an AI Model reference an AI Model Provider?
    a: |
      Set the `provider` field in each item of the [`targets`](/ai-gateway/entities/ai-model/#schema-aigateway-model-targets) array on the AI Model to the AI Model Provider's `name` or `id`.
---

## What is an AI Model Provider?

The AI Model Provider entity lets you securely store and manage credentials for connecting to upstream LLM services. Use AI Model Providers to:
* Store API keys for OpenAI, Azure, Bedrock, or any other LLM provider

> * Centrally manage and rotate credentials across multiple AI Models

Reviewer: ```suggestion
* Centrally manage and rotate credentials across multiple [AI Models](/ai-gateway/entities/ai-model/)
```

  - text: AI Policy entity

>     url: /ai-gateway/entities/ai-policy/

Author: @copilot fix it

## Manage AI Model Providers

AI Model Providers can be created and managed through:

* {{site.konnect_short_name}} UI
* {{site.ai_gateway}} API: `/v1/ai-gateways/{aiGatewayId}/model-providers`

For configuration examples and step-by-step setup instructions, see [Set up an AI Model Provider](#set-up-an-ai-model-provider) below.

### Relationship to AI Models

AI Model Providers and AI Models have a many-to-many relationship: one AI Model Provider can back many AI Models, and one AI Model can route to multiple AI Model Providers. For example, a single `openai` AI Model Provider might be used by both a chat AI Model and an embeddings AI Model, while a single AI Model might route to OpenAI and Anthropic targets for failover.

When configuring an [AI Model](/ai-gateway/entities/ai-model/), you reference an AI Model Provider by setting the `provider` field in each item of the [`targets`](/ai-gateway/entities/ai-model/#schema-aigateway-model-targets) array. You can reference by [`name`](#schema-aigateway-model-provider-name) or `id`. Use `id` if you plan to rename the AI Model Provider later.

## Supported AI Providers

> {{site.ai_gateway}} supports the following upstream AI providers. The AI Model Provider's [`type`](#schema-aigateway-model-provider-type) field selects one of these targets. The following AI provider-specific pages document supported capabilities, configuration requirements, and limitations.

Author: @copilot fix it

> ## Authentication

Reviewer: ```suggestion
## Outbound authentication
```
I think this is correct? This is how the Model Provider is authenticating with it's stored credentials with the LLM? I just got confused before and thought this was going to use the Identity Provider entity.

Author: Yes, this is outbound: so you authenticate to an outbound service with a Model Provider. Thanks!

    fallback: "`basic`"
  - type: "`azure`"
    providers: "[Azure OpenAI](/ai-gateway/ai-providers/azure/)"
    approach: "Microsoft Entra ID via Managed Identity (recommended when running in Azure). For explicit credentials, provide client ID, secret, and tenant ID. Requires `config.instance` (your Azure instance name, for example `kong-az-east`)."
    fallback: "`basic`"
  - type: "`gcp`"
    providers: "[Gemini](/ai-gateway/ai-providers/gemini/), [Vertex AI](/ai-gateway/ai-providers/vertex/)"
    approach: "Google service accounts via environment auto-detection (service account JSON or Compute Engine metadata server). For restricted networks, set `config.auth.metadata_url` or `config.auth.oauth_token_url` to custom endpoints."
    fallback: "`basic`"
{% endtable %}

## Lifecycle

Creating an AI Model Provider stores the entity but doesn't generate any runtime primitives. AI Model Provider credentials enter the runtime only when an AI Model references the AI Model Provider. At that point, the credentials are materialized into the underlying primitives of the AI Model.

> Updating an AI Model Provider re-materializes credentials into every AI Model that references it. The change takes effect on the next request through any referencing AI Model.

Reviewer: This feels too ephemeral and I don't have enough knowledge to suggest the changes myself. For example:
* "an AI Model Provider stores the entity", but we've established that AI Model Provider _is_ an entity. Should this be "stores the credentials"?
* "AI Model Provider credentials enter the runtime" should this be "are passed to the runtime"?
* "the credentials are materialized into the underlying primitives of the AI Model." The words "materialize" and "primitives" I think are giving the ephemeral effect here, I'm not really sure what's concretely happening.
* If we're passing the credentials, there are questions about how they are passed and stored. Are they encrypted in storage and in passing? 

Hmm, let me suggest a fix for what I think is happening.

```suggestion
An AI Model Provider stores the credentials, but doesn't generate any runtime primitives. 
AI Model Provider credentials are passed to the runtime only when an AI Model references the AI Model Provider. At that point, the credentials are then passed to the AI Model.

When you update a the credentials of an AI Model Provider, the new credentials are passed to every AI Model that references it the next time a request is made through the AI Model. 
```

Author: GitHub didn't let me apply your suggestion, so I had to do it manually. This makes sense!

## AI Policies and AI Model Providers

You can't attach [AI Policies](/ai-gateway/entities/ai-policy/) directly to an AI Model Provider entity instance. AI Policies attach to [AI Models](/ai-gateway/entities/ai-model/), [AI Agents](/ai-gateway/entities/ai-agent/), [AI Consumers](/ai-gateway/entities/ai-consumer/), or [AI Consumer Groups](/ai-gateway/entities/ai-consumer-group/) to control security, rate limiting, guardrails, and observability.

To apply an AI Policy across requests using a particular AI Model Provider, you can:

> 1. Set the policy to `global: true` to apply it to all resources in the gateway

Reviewer: ```suggestion
1. Set the policy to `global: true` to apply it to all resources in the gateway.
```

> 1. Attach the same policy to each AI Model that references the AI Model Provider

Reviewer: ```suggestion
1. Attach the same policy to each AI Model that references the AI Model Provider.
```

> 1. Create an AI Consumer Group with the policy and control access to AI Models via ACLs

Reviewer: ```suggestion
1. Create an AI Consumer Group with the policy and control access to AI Models via ACLs.
```

          value: Bearer <your-openai-key>
{% endentity_example %}

<!-- TODO: Uncomment before GA — kongctl declarative support for model_providers is not yet released.
```yaml
ai_gateways:
  - ref: <your-ai-gateway-ref>
    model_providers:
      - ref: my-openai-account
        name: my-openai-account
        display_name: OpenAI Production
        type: openai
        config:
          auth:
            type: basic
            header_name: Authorization
            header_value: Bearer <your-openai-key>
```

> -->

Reviewer: The link to the Konnect API page in the image is incorrect, it should point to `/konnect-api/`

<img width="721" height="234" alt="Image" src="https://github.com/user-attachments/assets/9f97bb0e-53f5-4d8c-a481-e35e833e8a41" />
